### PR TITLE
Add PyTorch FP16 benchmarks

### PR DESCRIPTION
**Pull request by mariecwhite**
Adds PyTorch BertLarge and ResNet50 FP16 benchmarks using the Inductor backend. Only about 50% of benchmarks are working due to errors in Inductor which is still experimental.

### CODE REVIEW

File: oobi/benchmark-definitions/python/pytorch_model_definitions.py [hunk 1]
```diff
@@ -30,6 +40,16 @@
     data_type=data_types.DataType.FP32,
 )
 
+BERT_LARGE_FP16_PT = data_types.MetaModel(
+    id=unique_ids.MODEL_BERT_LARGE_FP16_PT,
+    name="BERT_LARGE_FP16_PT",
+    tags=["fp32", "transformer-encoder", "bert"],
```
Comment (reichlfl) [inline oobi/benchmark-definitions/python/pytorch_model_definitions.py:46]:
```suggestion
    tags=["fp16", "transformer-encoder", "bert"],
```

File: iree-torch/benchmark/benchmark_model.py [hunk 2]
```diff
@@ -57,30 +57,47 @@ def bytes_to_mb_str(bytes: Optional[int]) -> str:
...
+        torch.set_default_tensor_type(torch.cuda.HalfTensor)
+        torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = True
+      elif torch_dtype == torch.float32:
+        torch.set_default_tensor_type(torch.cuda.FloatTensor)
+        torch.backends.cuda.matmul.allow_tf32 = True
+        torch.backends.cudnn.allow_tf32 = True
+      else:
+        raise ValueError(f"Datatype {data_type} not supported.")
+    elif backend == "cpu":
+      if torch_dtype != torch.float16:
```
Comment (reichlfl) [inline iree-torch/benchmark/benchmark_model.py:76]:
```suggestion
      if torch_dtype != torch.float32:
```

**Review by reichlfl — APPROVED** [review]

File: oobi/benchmark-definitions/python/pytorch_model_definitions.py (hunk 1 above)
Comment (mariecwhite) [reply oobi/benchmark-definitions/python/pytorch_model_definitions.py]: Thanks

File: iree-torch/benchmark/benchmark_model.py (hunk 2 above)
Comment (mariecwhite) [reply iree-torch/benchmark/benchmark_model.py]: Thanks for catching that!

**Pull request merged**